src/domain/accounts/domain/src/services/authentication_provider.rs

```diff
@@ -27,20 +29,15 @@ pub trait AuthenticationProvider: Sync + Send {
     async fn account_info_by_token(
         &self,
         provider_credentials_json: String,
-    ) -> Result<AccountInfo, InternalError>;
-
-    async fn find_account_info_by_name<'a>(
-        &'a self,
-        account_name: &'a AccountName,
-    ) -> Result<Option<AccountInfo>, InternalError>;
+    ) -> Result<Account, InternalError>;
 }
 
 ///////////////////////////////////////////////////////////////////////////////
 
 #[derive(Debug, Clone)]
 pub struct ProviderLoginResponse {
     pub provider_credentials_json: String,
-    pub account_info: AccountInfo,
+    pub account: Account,
```

Comment:
Did not expect to see `Account` here. Providers should return info about accounts in the first-party system based on which account in kamu can be created. IMO they shouldn't be deciding what identity to assign to an account, or whether it's `is_admin`.

Reply:
Improved. The root cause was starting in GraphQL code